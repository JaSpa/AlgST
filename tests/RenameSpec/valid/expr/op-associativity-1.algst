2 || 3 * 4 * negate 2 + global "a" () || 1
