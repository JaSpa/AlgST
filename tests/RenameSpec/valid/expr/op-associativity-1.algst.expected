Exp.App
|
+- Exp.App
|  |
|  +- Exp.Var (||)#Builtin.14
|  |
|  `- Exp.App
|     |
|     +- Exp.App
|     |  |
|     |  +- Exp.Var (||)#Builtin.14
|     |  |
|     |  `- Exp.Lit
|     |     |
|     |     `- Lit.Int 2
|     |
|     `- Exp.App
|        |
|        +- Exp.App
|        |  |
|        |  +- Exp.Var (+)#Builtin.22
|        |  |
|        |  `- Exp.App
|        |     |
|        |     +- Exp.App
|        |     |  |
|        |     |  +- Exp.Var (*)#Builtin.24
|        |     |  |
|        |     |  `- Exp.App
|        |     |     |
|        |     |     +- Exp.App
|        |     |     |  |
|        |     |     |  +- Exp.Var (*)#Builtin.24
|        |     |     |  |
|        |     |     |  `- Exp.Lit
|        |     |     |     |
|        |     |     |     `- Lit.Int 3
|        |     |     |
|        |     |     `- Exp.Lit
|        |     |        |
|        |     |        `- Lit.Int 4
|        |     |
|        |     `- Exp.App
|        |        |
|        |        +- Exp.Var negate#Builtin.35
|        |        |
|        |        `- Exp.Lit
|        |           |
|        |           `- Lit.Int 2
|        |
|        `- Exp.App
|           |
|           +- Exp.App
|           |  |
|           |  +- Exp.Var global#R.0
|           |  |
|           |  `- Exp.Lit
|           |     |
|           |     `- Lit.String "a"
|           |
|           `- Exp.Lit
|              |
|              `- Lit.Unit
|
`- Exp.Lit
   |
   `- Lit.Int 1

